internal/core/booking/reduce.go

```diff
@@ -77,6 +77,8 @@ func reduceAllAuthorReservationsByLongestPerSpot(reservations []*reservation.Res
 							reducedReservations = append(reducedReservations, r)
 						}
 					}
+				} else {
+					break
```

Comment:
Personally I'd enjoy a comment here; the reduce logic is already confusing for me, so "documenting" edge cases here would help a ton.